modules/core/shared/src/main/scala/retry/package.scala

```diff
@@ -146,35 +142,32 @@ package object retry:
       status: RetryStatus,
       a: A
   )(using
-      M: Monad[M],
-      S: Sleep[M]
+      T: Temporal[M]
   ): M[Either[RetryStatus, A]] =
-
     def onFalse: M[Either[RetryStatus, A]] = for
       nextStep <- applyPolicy(policy, status)
       _        <- onFailure(a, buildRetryDetails(status, nextStep))
       result <- nextStep match
         case NextStep.RetryAfterDelay(delay, updatedStatus) =>
-          S.sleep(delay) *>
-            M.pure(Left(updatedStatus)) // continue recursion
+          T.sleep(delay) *>
+            T.pure(Left(updatedStatus)) // continue recursion
         case NextStep.GiveUp =>
-          M.pure(Right(a)) // stop the recursion
+          T.pure(Right(a)) // stop the recursion
     yield result
 
     wasSuccessful(a).ifM(
-      M.pure(Right(a)), // stop the recursion
+      T.pure(Right(a)), // stop the recursion
       onFalse
     )
 
-  private def retryingOnSomeErrorsImpl[M[_], A, E](
+  private def retryingOnSomeErrorsImpl[M[_], A, E <: Throwable](
```

Comment:
Could remove the `E` type param here and hardcode it to `Throwable`? It's `Throwable` at all the call sites.

Reply:
Removed!